staging/src/k8s.io/kubectl/pkg/config/types.go

```diff
@@ -78,26 +78,28 @@ type AliasOverride struct {
 	// AppendArgs stores the arguments such as resource names, etc.
 	// These arguments are appended to the USER_ARGS.
 	AppendArgs []string
-	// Flag is allocated to store the flag definitions of alias
-	Flags []CommandOverrideFlag
+	// Options is allocated to store the option definitions of alias.
+	// Options only modify the default value of the option and if
+	// user explicitly passes a value, explicit one is used.
+	Options []CommandOptions
 }
 
-// CommandOverride stores the commands and their associated flag's
+// CommandDefaults stores the commands and their associated option's
 // default values.
-type CommandOverride struct {
+type CommandDefaults struct {
 	// Command refers to a command whose flag's default value is changed.
 	Command string
-	// Flags is a list of flags storing different default values.
-	Flags []CommandOverrideFlag
+	// Options is a list of options storing different default values.
+	Options []CommandOptions
 }
 
-// CommandOverrideFlag stores the name and the specified default
-// value of the flag.
-type CommandOverrideFlag struct {
-	// Flag name (long form, without dashes).
-	Name string `json:"name"`
+// CommandOptions stores the name and the specified default
+// value of an option.
+type CommandOptions struct {
```

Comment:
plural is a little odd here, maybe:
```suggestion
type CommandOptionDefault struct {
```

I'd also rename the v1alpha1 type so you don't need to hand-roll conversion

Reply:
> I'd also rename the v1alpha1 type so you don't need to hand-roll conversion

You're literally taking away all the pleasure from me :stuck_out_tongue_winking_eye: